kernel/aster-nix/src/syscall/pselect6.rs

```diff
@@ -11,17 +18,29 @@ pub fn sys_pselect6(
     timeval_addr: Vaddr,
     sigmask_addr: Vaddr,
 ) -> Result<SyscallReturn> {
-    // TODO: Support signal mask
-    if sigmask_addr != 0 {
-        error!("[SYS_PSELECT6] Not support sigmask now");
-        return Err(Error::new(Errno::ENOSYS));
-    }
+    let current_thread = current_thread!();
+    let posix_thread = current_thread.as_posix_thread().unwrap();
+
+    let old_simask = if sigmask_addr != 0 {
+        let new_sigmask: SigMask = read_val_from_user(sigmask_addr)?;
+        let old_sigmask = posix_thread.sig_mask().swap(new_sigmask, Ordering::Release);
```

Comment:
No need for release ordering here, `Ordering::Relaxed` would be fine, as 1) other thread may only read this value and 2) they do not care about the precise time the mask is changed, they just use the value for a heuristic.